docs/en/stack/data-frames/dataframe-examples.asciidoc

```diff
@@ -0,0 +1,362 @@
+        "bucket_script": {
+          "buckets_path": {
+            "delay_time": "delay_mins_total.value",
+            "flight_time": "flight_mins_total.value"
+          },
+          "script": "(params.delay_time / params.flight_time) * 100"
+        }
+      }
+    }
+  }
+}
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+<1> Filter the source data to select only flights that were not cancelled.
+<2> The data is grouped by the `Carrier` field which contains the airline name.
+<3> This `bucket_script` performs calculations on the results that returned by 
+the aggregation, in this particular example to calculate what percentage of 
+travel time was taken up by delays.
+
+The API returns the following response:
+
+[source,js]
+----------------------------------
+{
+  "preview" : [
+    {
+      "carrier" : "ES-Air",
+      "flights_count" : 2802.0,
+      "flight_mins_total" : 1436927.5130677223,
+      "delay_time_percentage" : 9.335543983955839,
+      "delay_mins_total" : 134145.0
+    },
+    {
+      "carrier" : "JetBeats",
+      "flights_count" : 2833.0,
+      "flight_mins_total" : 1451143.6898144484,
+      "delay_time_percentage" : 8.937088787987832,
+      "delay_mins_total" : 129690.0
+    },
+    {
+      "carrier" : "Kibana Airlines",
+      "flights_count" : 2832.0,
+      "flight_mins_total" : 1419081.404241085,
+      "delay_time_percentage" : 9.088273556017194,
+      "delay_mins_total" : 128970.0
+    },
+    {
+      "carrier" : "Logstash Airways",
+      "flights_count" : 2914.0,
+      "flight_mins_total" : 1503620.8713908195,
+      "delay_time_percentage" : 9.544959286661593,
+      "delay_mins_total" : 143520.0
+    }
+  ]
+}
+----------------------------------
+// CONSOLE
+// TEST[skip:SETUP]
+
+The example above transforms flight data into a entity centric index for 
+flight carriers. This makes it easier to answer questions such as:
+
+* Which air carrier has the most delays as a percentage of flight time?
+
+NOTE: Please note that this data is fictional and does not reflect actual delays 
+or flight stats for any of the featured destination or origin airports.
+
+[float]
+[[example-clientips]]
+=== Finding suspicious client IPs by using scripted metrics
+
+With {dataframe-transforms}, you can use 
+{ref}/search-aggregations-metrics-scripted-metric-aggregation.html[scripted 
+metric aggregations] on your data. These aggregations are flexible and make 
+it possible to perform very complex processing. Let's use scripted metrics to 
+identify suspicious client IPs in the web log sample dataset.
+
+We transform the data such that the new index contains the sum of bytes and the 
+number of distinct URLs, agents, incoming requests by location, and geographic 
+destinations for each client IP. We also use a scripted field to count the 
+specific types of HTTP responses that each client IP receives. Ultimately, the 
+example below transforms web log data into an entity-centric index where the 
```

Comment:
Here "entity-centric" has a hyphen and in other cases on this page it does not.  We should be consistent one way or the other.